### PR TITLE
[JEP-7] Comment updates after removing JRuby support

### PR DESCRIPTION
**Pull request by jglick**
Amending #6209.

### Proposed changelog entries

* N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by MarkEWaite** [comment]
Won't this break the ivy plugin, based on:

https://github.com/jenkinsci/ivy-plugin/blob/5a41f158856e401fc7dd07d95459735757bb5539/src/main/java/hudson/ivy/SplittableBuildListener.java#L48

**Comment by jglick** [comment]
> Won't this break the ivy plugin

What would break? `AbstractTaskListener` was long since deprecated; here I am just removing its use from some core impls.

Originally copied from `maven-plugin` BTW; the original was already cleaned up in https://github.com/jenkinsci/maven-plugin/pull/112. Which reminds me that this PR basically reverts #3154, going back to #3122. See also https://issues.jenkins.io/browse/JENKINS-48116 & https://issues.jenkins.io/browse/JENKINS-50124.

**Comment by jglick** [comment]
Correct, so this should not pose any problem for Jenkins users. I need to check with @jtnord about #3346 and whether this could break proprietary functionality, which apparently cares about cross-version serial form of `StreamTaskListener`.

**Review by timja — DISMISSED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
(checkstyle failures)

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
